augurpy/estimators.py

```diff
@@ -1,25 +1,57 @@
-from typing import Dict, List, Optional, Union
+from typing import Dict, Literal, Union
 
 from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
 from sklearn.linear_model import LogisticRegression
+from typing_extensions import TypedDict
+
+
+class Params(TypedDict, total=False):
+    """Type signature for random forest and logistic regression parameters."""
+
+    n_estimators: int
+    max_depth: int
+    max_features: Union[Literal["auto"], Literal["log2"], Literal["sqrt"], int, float]
+    penalty: Union[Literal["l1"], Literal["l2"], Literal["elasticnet"], Literal["none"]]
```

Comment:
```suggestion
    penalty: Union[Literal["l1"], Literal["l2"], Literal["elasticnet"], None
```

Think this should also work? We can handle None.

Reply:
Done.

Reply:
Ah but this goes directly into the [logistic regression](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html?highlight=logistic%20regression#sklearn.linear_model.LogisticRegression) as a parameter. It needs 'none' not None, so I would rather keep the string, unless Im missing something.

Reply:
Huh, didn't remember this. But you are right. Let's do it like you had it!